File: app/src/main/java/com/example/mapboxdemo/SignalRListener.java
package com.example.mapboxdemo;

import android.util.Log;

import com.microsoft.signalr.HubConnection;
import com.microsoft.signalr.HubConnectionBuilder;

public class SignalRListener {
    private static SignalRListener instance;

    HubConnection hubConnection;

    public SignalRListener() {
        Log.i("TAG", "Constructor");
        hubConnection = HubConnectionBuilder.create("http://192.168.0.169:5005/update").build();
        hubConnection.on("UpdateMarker", (markerId) -> {
                MainActivity.instance.updateMarkerById(markerId);
        }, Integer.class);
        hubConnection.on("DeleteMarker", (markerId) -> {
            MainActivity.instance.removeMarkerById(markerId);
        }, Integer.class);
        hubConnection.on("AddMarker", (markerId) -> {
            Log.i("TAG", "AddMarker");
            MainActivity.instance.requestMarkerById(markerId);
        }, Integer.class);
    }

    public static SignalRListener getInstance() {
        if (instance == null)
            instance = new SignalRListener();
        return instance;
    }

    public void startConnection() {
        Log.i("TAG", "Starting hub connection");
        hubConnection.start();
    }

    public void stopConnection() {
        Log.i("TAG", "Stopping hub connection");
        hubConnection.stop();
    }
}

File: app/src/main/java/com/example/mapboxdemo/SignalRListener.kt
package com.example.mapboxdemo

import android.util.Log
import com.microsoft.signalr.HubConnection
import com.microsoft.signalr.HubConnectionBuilder

class SignalRListenerK {
    private var hubConnection : HubConnection =
        HubConnectionBuilder.create("https://192.168.0.169:5006/update").build()

    init {
        hubConnection.on("UpdateMarker", { markerId ->
            MainActivity.instance.updateMarkerById(markerId)
        }, Int::class.java)
        hubConnection.on("DeleteMarker", { markerId ->
            MainActivity.instance.removeMarkerById(markerId)
        }, Int::class.java)
        hubConnection.on("AddMarker", { markerId ->
            Log.i("TAG", "AddMarker")
            MainActivity.instance.requestMarkerById(markerId)
        }, Int::class.java)
    }

    fun startConnection() {
        Log.i("TAG", "Started hub connection")
        hubConnection.start()
    }

    fun stopConnection() {
        hubConnection.stop()
    }
}